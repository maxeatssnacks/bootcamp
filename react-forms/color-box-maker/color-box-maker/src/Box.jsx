import React from "react";

const Box = ({ id, color, width, height }) => {
    const boxStyle = {
        width: `${width}px`,
        height: `${height}px`,
        backgroundColor: color
    }

    return (
        <div id={id} style={boxStyle}></div>
    )
}

export default Box;