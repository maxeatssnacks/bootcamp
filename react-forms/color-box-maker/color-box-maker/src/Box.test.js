import React from 'react';
import { render } from '@testing-library/react';
import Box from './Box';

describe('Box', () => {
    // Smoke test
    test('renders without crashing', () => {
        render(<Box id="1" color="#FF0000" width={100} height={100} />);
    });

    // Snapshot test
    test('matches snapshot', () => {
        const { asFragment } = render(<Box id="1" color="#FF0000" width={100} height={100} />);
        expect(asFragment()).toMatchSnapshot();
    });

    // Critical business logic test
    test('has correct style properties', () => {
        const { container } = render(<Box id="1" color="#FF0000" width={100} height={100} />);
        const box = container.firstChild;

        expect(box).toHaveStyle('background-color: #FF0000');
        expect(box).toHaveStyle('width: 100px');
        expect(box).toHaveStyle('height: 100px');
    });
});