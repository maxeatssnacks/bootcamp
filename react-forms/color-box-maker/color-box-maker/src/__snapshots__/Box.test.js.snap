// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Box matches snapshot 1`] = `
<DocumentFragment>
  <div
    id="1"
    style="width: 100px; height: 100px; background-color: rgb(255, 0, 0);"
  />
</DocumentFragment>
`;
